crates/mvr-cli/src/utils/api_data.rs

```diff
@@ -87,6 +90,48 @@ pub async fn query_multiple_dependencies(
     Ok(package_requests)
 }
 
+pub async fn search_names(
+    search: Option<String>,
+    limit: Option<u32>,
+    cursor: Option<String>,
+) -> Result<SearchNamesResponse> {
+    let mut params = HashMap::new();
+    params.insert("limit", limit.unwrap_or(DEFAULT_LIMIT).to_string());
+
+    if let Some(search) = search {
+        params.insert("search", search);
+    }
+
+    if let Some(cursor) = cursor {
+        params.insert("cursor", cursor);
+    }
+
+    let client = Client::new();
+
+    let response = client
+        .get(format!("{}/v1/names", get_api_url(&Network::Mainnet)?,))
+        .query(&params)
+        .send()
+        .await
+        .map_err(|e| {
+            anyhow::anyhow!(
+                "Failed to search names: {}. Error: {}",
+                params.get("search").unwrap_or(&String::default()),
```

Comment:
Not sure this would work, but `&String::default` looks a bit strange to me
```suggestion
                params.get("search").unwrap_or_default(),
```

Reply:
Yeah, it expected `&String`, following what `.get()` returns, however, I did make this look less bad by cloning its value & unwrapping_or_default().